client/ayon_houdini/api/plugin.py

```diff
@@ -325,7 +326,27 @@ def filepath_from_context(cls, context):
         if cls.use_ayon_entity_uri:
             return get_ayon_entity_uri_from_representation_context(context)
 
-        return super(HoudiniLoader, cls).filepath_from_context(context)
+        path = super().filepath_from_context(context)
+
+        # Remap project roots to the relevant environment variables
+        # TODO: Refactor this to more optimal/better code
+        mapping = {}
+        for key, value in os.environ.items():
+            if key.startswith("AYON_PROJECT_ROOT_"):
+                mapping[key] = value
+        if mapping:
+            match_path = path.replace("\\", "/")
+            # Sort by length to ensure that the longest matching key is first
+            # so that the nearest matching root is used
+            for key, value in sorted(mapping.items(),
+                                     key=lambda x: len(x[1]),
+                                     reverse=True):
+                if value and match_path.startswith(value.replace("\\", "/")):
+                    # Replace start of string with the key
+                    path = f"${key}" + path[len(value):]
+                    break
```

Comment:
I wonder if going the other way is easier to understand and maintain. 
I'm assuming the longest match will result in a smaller string (unless the root name is big enough to break this assumption)
Also, we can skip sorting if we the matching is one path.
```suggestion
        paths = []
        for key, value in mapping.items():
            if value and path.startswith(value.replace("\\", "/")):
                # Replace start of string with the key
                paths.append(f"${key}{path[len(value):]}")

        if len(paths) == 1:
            return paths[0]

        # sort and return the smallest path.
        return sorted(paths)[0]
```

Reply:
I like this less because it'd mean we'd always be checking against all roots, even if we had already matched one. So it'd perform less optimal. However, sure - the amount of roots may never grow that large that it'd really be a problem.

Overall I just did a prototype here and did not look at 'cleaning the code' at all - it was merely testing whether this _could_ be functional.

Reply:
> the amount of roots may never grow that large that it'd really be a problem.

I agree. but, the code just looks hard to read that's why I tried picking a different approach.